### PR TITLE
HV: fix "use -- or ++ operations"

### PR DESCRIPTION
**Pull request by huix-shh**
ACRN coding guidelines banned -- or ++ operations.
V1->V2:
  add comments to struct stack_frame

Tracked-On: #861
Signed-off-by: Huihuang Shi <huihuang.shi@intel.com>

### CODE REVIEW

**Comment by HuiHuangShi** [comment]
3 new potential violations to the coding guideline detected. Please help check. Thanks!<br><1> config:common filename:/hypervisor/arch/x86/guest/vcpu.c func: offset:0<br> reason:Use of single line comment(s) //.<br><2> config:common filename:/hypervisor/arch/x86/guest/vcpu.c func:build_stack_frame offset:701<br> reason:Cast from integral type to pointer. : (unsigned long to struct stack_frame*): ( struct stack_frame * ) rsp<br><3> config:common filename:/hypervisor/arch/x86/guest/vcpu.c func:build_stack_frame offset:716<br> reason:Cast from pointer to integral type. : (unsigned long* to unsigned long): ( uint64_t ) ret<br><br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

File: hypervisor/arch/x86/guest/vcpu.c [hunk 1]
```diff
@@ -680,23 +694,26 @@ void schedule_vcpu(struct acrn_vcpu *vcpu)
...
-	*sp-- = 0UL; /* rbx */
-	*sp-- = 0UL; /* rbp */
-	*sp-- = 0UL; /* r12 */
-	*sp-- = 0UL; /* r13 */
-	*sp-- = 0UL; /* r14 */
-	*sp-- = 0UL; /* r15 */
-	*sp = (uint64_t)&vcpu->sched_obj; /*rdi*/
-
-	return (uint64_t)sp;
+	sp = (struct stack_frame *)rsp;
```
Comment (dongyaozu) [inline hypervisor/arch/x86/guest/vcpu.c]: should we do:  sp = sp -1 here?

File: hypervisor/arch/x86/guest/vcpu.c [hunk 2]
```diff
@@ -680,23 +694,26 @@ void schedule_vcpu(struct acrn_vcpu *vcpu)
 static uint64_t build_stack_frame(struct acrn_vcpu *vcpu)
 {
 	uint64_t rsp = (uint64_t)&vcpu->stack[CONFIG_STACK_SIZE - 1];
-	uint64_t *sp;
+	struct stack_frame *sp;
```
Comment (dongyaozu) [inline hypervisor/arch/x86/guest/vcpu.c]: rsp means the register of stack here? seems not too. It mean the stack top?
sp is a little bit confusing. It means stack frame pointer, right?  how about sfp? or frame or sf?

**Comment by HuiHuangShi** [comment]
2 new potential violations to the coding guideline detected. Please help check. Thanks!<br><1> config:common filename:/hypervisor/arch/x86/guest/vcpu.c func:build_stack_frame offset:701<br> reason:Cast from integral type to pointer. : (unsigned long to struct stack_frame*): ( struct stack_frame * ) rsp<br><2> config:common filename:/hypervisor/arch/x86/guest/vcpu.c func:build_stack_frame offset:716<br> reason:Cast from pointer to integral type. : (unsigned long* to unsigned long): ( uint64_t ) ret<br><br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

File: hypervisor/arch/x86/guest/vcpu.c [hunk 3]
```diff
@@ -677,26 +691,32 @@ void schedule_vcpu(struct acrn_vcpu *vcpu)
 	release_schedule_lock(vcpu->pcpu_id);
 }
 
+/*
+* pre &vcpu->stack[CONFIG_STACK_SIZE] shall align to 16
```
Comment (dongyaozu) [inline hypervisor/arch/x86/guest/vcpu.c]: Is this a valid @pre? Please check with Xiangyang?

**Comment by HuiHuangShi** [comment]
3 new potential violations to the coding guideline detected. Please help check. Thanks!<br><1> config:common filename:/hypervisor/arch/x86/guest/vcpu.c func:build_stack_frame offset:699<br> reason:Array bound exceeded.                        : stack[*]; accessed=8192, range=0-8191<br><2> config:common filename:/hypervisor/arch/x86/guest/vcpu.c func:build_stack_frame offset:703<br> reason:Cast from integral type to pointer. : (unsigned long to struct stack_frame*): ( struct stack_frame * ) stacktop<br><3> config:common filename:/hypervisor/arch/x86/guest/vcpu.c func:build_stack_frame offset:719<br> reason:Cast from pointer to integral type. : (unsigned long* to unsigned long): ( uint64_t ) ret<br><br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

File: hypervisor/arch/x86/guest/vcpu.c (hunk 3 above)
Comment (wuxyintel) [reply hypervisor/arch/x86/guest/vcpu.c]: Only need to update format about pre-condition, this pre-condition is guaranteed by compiler and related attribute (__aligned).

**Comment by HuiHuangShi** [comment]
3 new potential violations to the coding guideline detected. Please help check. Thanks!<br><1> config:common filename:/hypervisor/arch/x86/guest/vcpu.c func:build_stack_frame offset:699<br> reason:Array bound exceeded.                        : stack[*]; accessed=8192, range=0-8191<br><2> config:common filename:/hypervisor/arch/x86/guest/vcpu.c func:build_stack_frame offset:703<br> reason:Cast from integral type to pointer. : (unsigned long to struct stack_frame*): ( struct stack_frame * ) stacktop<br><3> config:common filename:/hypervisor/arch/x86/guest/vcpu.c func:build_stack_frame offset:719<br> reason:Cast from pointer to integral type. : (unsigned long* to unsigned long): ( uint64_t ) ret<br><br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Pull request merged**